src/device-registry/models/Device.js

```diff
@@ -259,6 +259,25 @@ deviceSchema.statics = {
       logObject("the args", args);
       logger.info("in the register static fn of the Device model...");
       let modifiedArgs = args;
+
+      if (
+        isEmpty(modifiedArgs.name) &&
+        !isEmpty(args.generation_version) &&
+        !isEmpty(args.generation_count)
+      ) {
```

Comment:
When creating a device name, only use the generation count and generation number if they are provided.

Otherwise, just go ahead and some small sanitisation of the provided name.